src/utils/nextPurchaseDay.js

```diff
@@ -0,0 +1,45 @@
+import { diffBetweenTodayAndDate } from './diffBetweenTodayAndDate';
+/***
+ * First we verify by specificity and
+ * we return the string that is used for the
+ * aria-label and to add the proper style
+ */
+
+export const nextPurchaseDay = (
+  daysUntilNextPurchase,
+  lastPurchaseDate,
+  numberOfPurchases,
+) => {
+  /**
+   * After some checks, we noticed lastPurchasedDate is a Date object
+   * and must be transformed to a day in order to properly compare
+   * However, there are some cases where the item has not been bought yet
+   * such as when the item has just been added to the list
+   * in this case, the lastPurchaseDate value is undefined
+   * and to avoid that error, we give a new Date() value
+   *
+   * After that, we can use diffBetweenTodayAndDate function to get
+   * the difference in days between lastPurchaseDate and today.
+   **/
+  const lastPurchase = lastPurchaseDate
+    ? lastPurchaseDate.toDate()
+    : new Date();
+  const daysSinceLastPurchase = diffBetweenTodayAndDate(lastPurchase);
+  const nextEstimatedPurchase = 2 * daysUntilNextPurchase;
+
+  if (
+    numberOfPurchases === 1 ||
+    daysSinceLastPurchase >= nextEstimatedPurchase
+  ) {
+    return 'inactive';
+  }
+  if (daysUntilNextPurchase < 7) {
+    return 'soon';
+  }
+  if (daysUntilNextPurchase >= 7 && daysUntilNextPurchase <= 30) {
+    return 'kindOfSoon';
+  }
+  if (daysUntilNextPurchase > 30) {
+    return 'notAnySoon';
+  }
+};
```

Comment:
In a quick view, it makes sense to me 👍 . On another hand, I'm honest I couldn't check it because I left some comments in Slack about a small bug. Of course, If I'm wrong with this change, I'll appreciate your comments and understand why this is the correct value 😄 .

![bug](https://user-images.githubusercontent.com/8689897/141579617-3bff0d4f-830a-4f77-830a-55c927c09c49.png)

Reply:
Confirm, your conditionals to sort the products are OK 👍 .